### PR TITLE
feat(meeting): add mute participant controls

### PR DESCRIPTION
**Pull request by chrisadubois**
# COMPLETES # https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-393351

## This pull request addresses

Adding mute on entry and disallow unmute options to the meeting

## by making the following changes

New controls options manager class + proxies and listeners

### Change Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Tooling change
- [ ] Internal code refactor

## The following scenarios where tested

Host/Non host on converged meeting (manual), with and without native client
unit test

### I certified that

- [x] I have read and followed [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request)
- [ ] I discussed changes with code owners prior to submitting this pull request
- [x] I have not skipped any automated checks
- [x] All existing and new tests passed
- [x] I have updated the documentation accordingly

### CODE REVIEW

File: packages/@webex/plugin-meetings/src/controls-options-manager/index.ts [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+   * @param {Setting} setting
+   * @param {boolean} enabled
+   * @private
+   * @memberof ControlsOptionsManager
+   * @returns {Promise}
+   */
+  private setControls(setting: Setting, enabled: boolean): Promise<any> {
+    LoggerProxy.logger.log(`ControlsOptionsManager:index#setControls --> ${setting} [${enabled}]`);
+
+    if (Util?.[`${enabled ? Setting.canSet : Setting.canUnset}${setting}`](this.displayHints)) {
```
Comment (chburket) [inline packages/@webex/plugin-meetings/src/controls-options-manager/index.ts:145]: question (non-blocking): can Util be undefined here?  Also, I personally think this line is a little hard to parse visually.  Not sure how to make it more obvious what's going on, though.  Maybe a Util.canSet(setting, displayHints)?

File: packages/@webex/plugin-meetings/test/unit/spec/controls-options-manager/index.js [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+                it('can set and extract new values later on', () => {
+                    const manager = new ControlsOptionsManager({});
+                    assert.isUndefined(manager.getLocusUrl());
+                    manager.set({
+                      locusUrl: 'test/id',
+                    })
+                    assert(manager.getLocusUrl(), 'test/id');
+                });
+            });
+
```
Comment (chburket) [inline packages/@webex/plugin-meetings/test/unit/spec/controls-options-manager/index.js:21]: nit: extra line

File: packages/@webex/plugin-meetings/test/unit/spec/controls-options-manager/util.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+                locusInfo.parsedLocus.info.userDisplayHints.push('DISABLE_MUTE_ON_ENTRY');
+
+                assert.equal(ControlsOptionsUtil.canUnsetMuteOnEntry(locusInfo.parsedLocus.info.userDisplayHints), true);
+              });
+        
+              it('rejects when correct display hint is not present', () => {
+                assert.equal(ControlsOptionsUtil.canSetMuteOnEntry(locusInfo.parsedLocus.info.userDisplayHints), false);
+              });
+
+              it('rejects when correct display hint is not present', () => {
```
Comment (chburket) [inline packages/@webex/plugin-meetings/test/unit/spec/controls-options-manager/util.js:37]: nit: these have the same name

**Review by chburket — APPROVED** [review]
Looks good!  Nothing blocking, but ping me if you make any changes based on my comments and I'll rereview.

File: packages/@webex/plugin-meetings/src/controls-options-manager/index.ts [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+      // @ts-ignore
```
Comment (cscheffauer) [inline packages/@webex/plugin-meetings/src/controls-options-manager/index.ts:146]: why ts-ignore here?

**Review by cscheffauer — APPROVED** [review]
lgtm! small comment on the ts-ignore and waiting for the comments from Charles to be addressed.

File: packages/@webex/plugin-meetings/src/controls-options-manager/index.ts [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+  }
+
+  /**
+   * @param {Setting} setting
+   * @param {boolean} enabled
+   * @private
+   * @memberof ControlsOptionsManager
+   * @returns {Promise}
+   */
+  private setControls(setting: Setting, enabled: boolean): Promise<any> {
```
[the remark this answers is not in the document]
Comment (chrisadubois) [reply packages/@webex/plugin-meetings/src/controls-options-manager/index.ts:142]: It will have more controls !

File: packages/@webex/plugin-meetings/src/controls-options-manager/enums.ts [hunk 6]
```diff
@@ -0,0 +1,9 @@
+enum Setting {
+  disallowUnmute = 'DisallowUnmute',
+  muteOnEntry = 'MuteOnEntry',
+  enabled = 'enabled',
+  canSet = 'canSet',
+  canUnset = 'canUnset',
+}
```
[the remark this answers is not in the document]
Comment (chrisadubois) [reply packages/@webex/plugin-meetings/src/controls-options-manager/enums.ts:7]: Ah I suppose you're right.

File: packages/@webex/plugin-meetings/src/controls-options-manager/index.ts (hunk 1 above)
Comment (chrisadubois) [reply packages/@webex/plugin-meetings/src/controls-options-manager/index.ts:145]: I was thinking the same thing, and kind of either way it's a bit messy. I could clean it up a touch though.

File: packages/@webex/plugin-meetings/src/controls-options-manager/index.ts (hunk 4 above)
Comment (chrisadubois) [reply packages/@webex/plugin-meetings/src/controls-options-manager/index.ts:146]: You have to because of the typescript thing that happened with plugin-meetings. I think Tim or Arun might be working on this

File: packages/@webex/plugin-meetings/src/controls-options-manager/index.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (chrisadubois) [reply packages/@webex/plugin-meetings/src/controls-options-manager/index.ts:142]: and also the API is named controls
